strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/LayoutFileSystem.java

```diff
@@ -32,11 +32,11 @@
     public static final String TEMP = ".temp";
     public static final String INDEX = ".index";
 
-    private final Repository repository;
+    private final RepositoryData repository;
     private final StorageFileSystemProvider provider;
     
     public LayoutFileSystem(PropertiesBooter propertiesBooter,
-                            Repository repository,
+                            RepositoryData repository,
```

Comment:
I'm not sure how we should go about this...

If you refactor it, then I guess the variable names will also have to be renamed as well.

What are your thoughts on this @sbespalov , @fuss86 , @steve-todorov ...?

Reply:
@carlspring variable names shouldn't be renamed, because in this case this is actually a repository.
`RepositoryData` is just an interface which defines what data(fields) the repository has. There is also an immutable implementation with final fields called just `Repository` and mutable implementation with getters/setters called `RepositoryDto`, so all variables of this types can be called just `repository`. 
Actually we have this concept while ago, and it will be the same for `StorageData` etc.

Within this PR I have just make all of such entities to follow same naming concept and use `*Data` suffix for interfaces instead of `*ReadContract` which sounds much better I think.

Reply:
@carlspring maybe if we will swap the names it will be more pleasant for you? :smiley: 
I mean it can be like follows:
- `Repository` will be the interface name
- `RepositoryData` will be the immutable class name
- `RepositryDto` will be the mutable class name

this way it will be more like `List` and `ArrayList`.

Reply:
I think that this is a valid point! Let's do it like this.

Reply:
done